lib/fibonacci.rb

```diff
@@ -1,8 +1,21 @@
 # Improved Fibonacci
 
-# Time Complexity - ?
-# Space Complexity - ?  (should be O(n))
+# Time Complexity - O(n)
+# Space Complexity - O(n) (should be O(n))
 # Hint, you may want a recursive helper method
 def fibonacci(n)
```

Comment:
Consider how you can do this with only a 2-element array.  That would have the same space/time complexity, but it would be a little more efficient.